gfx/common/x11_common.c

```diff
@@ -306,9 +306,17 @@ static void x11_suspend_screensaver_xdg_screensaver(Window wnd, bool enable)
 
 void x11_suspend_screensaver(Window wnd, bool enable)
 {
-    x11_suspend_screensaver_xdg_screensaver(wnd, enable);
 #ifdef HAVE_DBUS
     x11_suspend_screensaver_dbus(enable);
+    if (dbus_screensaver_cookie == 0)
+    {
+        // Only attempt xdg-screensaver method if dbus didn't succeed
+#endif
+
+        x11_suspend_screensaver_xdg_screensaver(wnd, enable);
+
+#ifdef HAVE_DBUS
+    }
```

Comment:
> Ah, but I think it is

Then I'm operating on stale data. Sorry about that.

Still, x11_suspend_screensaver should work in both directions, even if we don't use both directions right now; anything else is going to create bugs later. (It technically works right now, but for wrong reasons, and will confuse anyone else looking at that thing.) Add a || enable and I'll accept it. (Or || !enable if enable==true means enable suspension... that entire function family should be renamed x11_set_allow_screensaver, the double negation annoys me.)

> Other functions have been updated with similar blocks to call DBus-specific code

They call dbus functions, but don't poke dbus variables directly.

...whatever, it works, I've seen worse things in RetroArch. It's good enough. We can clean it up later if it becomes a problem.